### PR TITLE
Adds .github/CODEOWNERS file

### PR DESCRIPTION
**Pull request by Cyberboss**
https://help.github.com/articles/about-codeowners/

Basically it's Github's way of subscribing to file changes. You put your name on the list and you get requested when a PR that touches the files is opened.

### CODE REVIEW

**Comment by RemieRichards** [comment]
tag me on shit I put my name on @Cyberboss k thx.

**Comment by MrStonedOne** [comment]
add me for the mc core files as well as sstimer and ssair, and the atmos module

**Comment by MoreRobustThanYou** [comment]
Add my name to RCL.dm

**Comment by Cyberboss** [comment]
Sorry @MoreRobustThanYou, it only works on org memebers

**Comment by MoreRobustThanYou** [comment]
oh, ok

**Comment by Cyberboss** [comment]
@tgstation/commit-access any other requests?

**Comment by Cyberboss** [comment]
@MoreRobustThanYou At least, that was the case with my tests, I'll run a trial run with you, if it works, then cool

**Comment by lzimann** [comment]
i don't want any formal obligations

**Comment by Tacolizard** [comment]
@Cyberboss if @MoreRobustThanYou works with this, then please add me on all the organ and carbon/human life files.

can you do this with an entire directory?

**Comment by optimumtact** [comment]
Considering I cannot request a review from them under normal circumstances I doubt it works

**Comment by optimumtact** [comment]
On the whole I don't agree with this, there is no single code owner, and I expect everyone to own it all equally, this gives people a false impression

**Comment by Cyberboss** [comment]
It's a terrible title for a useful feature

**Comment by MoreRobustThanYou** [comment]
^

**Comment by Tacolizard** [comment]
goof gets in but i dont 😢

**Comment by Cyberboss** [comment]
All non-org people on the list will just get removed anyway if it doesn't work (and I'm fairly certain it won't)

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+/code/datums/forced_movement.dm @Cyberboss
+/code/datums/holocall.dm @Cyberboss
+/code/modules/admin/verbs/adminhelp.dm @Cyberboss
+/code/modules/admin/verbs/adminpm.dm @Cyberboss
+/code/modules/server_tools/* @Cyberboss
+/code/modules/mapping/* @Cyberboss
+
+# Iamgoofball
+
+/code/modules/GOAP/* @Iamgoofball
```
Comment (RemieRichards) [inline .github/CODEOWNERS]: considering I wrote the GOAP framework, you should put me on it too (and really only put goof on the actual examples like GOAP medibots, GOAP use cases etc. cos he made those)

**Comment by JJRcop** [comment]
If we decide to honestly do this, people can be added to the org as third party contributors if needed.

**Comment by RemieRichards** [comment]
I doubt it, there are very few cases where someone will produce something Code Owner worthy and NOT be a candidate for being on the team anyway.

That's talking of course, if it IS limited to org members.

**Comment by lzimann** [comment]
Can't see why it wouldn't be limited to org members. It probably uses the same system as manually requesting a reviewer. But then again, github sometimes is weird.

**Comment by RemieRichards** [comment]
yeah I think everyone is assuming that.
would be interesting if it did work though.

**Comment by Iamgoofball** [comment]
at this point it's a 50/50 split between goofcode and remiecode in the goap framework

why does it have to be so drastic

cant we just have weekend codeownership alternating?

**Comment by RemieRichards** [comment]
or we could have the parts we each wrote
like I said
(you getting the examples, the actual uses of the framework, and me getting the framework)

**Comment by kyrahabattoir** [comment]
Can you tag me to:

* code/game/objects/effects/spawners/lootdrop.dm
* code/game/objects/items/weapons/vending_items.dm
* code/game/objects/effects/spawners/structure.dm

**Comment by Cyberboss** [comment]
OK no more non-org tags until we confirm it works

**Comment by Iamgoofball** [comment]
but I also had to rework and bugfix the framework and do a lot of maint stuff and also make it work since it didn't work when I started

what if we arrange code ownership visitation rights on a biweekly schedule

**Comment by JJRcop** [comment]
If non-org works then both goof and rr can be added to goap

**Comment by optimumtact** [comment]
why this is a bad idea

**Comment by JJRcop** [comment]
@RemieRichards @Iamgoofball I think this file is best used for you to get notified when someone changes a system that you know very well, so that you can review their code and make sure they did it right, see how MSO wants to be notified of MC, timers, and atmos changes, because he knows how they all work very well, and his input on any pull request regarding those is very valuable.

I think it's unhealthy to look at this as owning the code, even though it's called codeowners, that's not what /tg/station is about.

See this excerpt from [CONTRIBUTING.md](https://github.com/tgstation/tgstation/blob/fedccea51931527a67405e14090a073bc2294a9d/.github/CONTRIBUTING.md)

>If your pull request is accepted, the code you add no longer belongs exclusively to you but to everyone; everyone is free to work on it, but you are also free to support or object to any changes being made, which will likely hold more weight, as you're the one who added the feature. It is a shame this has to be explicitly said, but there have been cases where this would've saved some trouble.

**Comment by Cyberboss** [comment]
I guarantee it's not going to work for non-org members anyway, I'm just humoring them

**Comment by RemieRichards** [comment]
@JJRcop yes, exactly, that's the entire point.
I wrote the GOAP framework, so I'm in the best position to review it.
Goof wrote the GOAP Use cases (Lavaland Russians, Medibots, Miners etc.) so he's in the best position to review those.

I resent what you're trying to imply.

**Comment by optimumtact** [comment]
you're literally proving my point

**Comment by RemieRichards** [comment]
How so, do you deny that the author of a work is the one most capable of reviewing it? if so I'll value your opinion even less in future.

**Comment by optimumtact** [comment]
[](http://www.quickmeme.com/img/eb/eb6140942424697bf997986aac9b2d682cd18a18fe192e4f81ba32f17b629a24.jpg)

**Comment by kyrahabattoir** [comment]
Can I be made an org member?

**Comment by optimumtact** [comment]
your avatar is frankly terrifying

**Comment by MrStonedOne** [comment]
lets just merge this, and let the infighting happen on their own prs.

also, two people can get notified over the same file.

also, it being repo org members only is lame. maybe somebody should just make a bot where people can subscribe to code changes and the bot just mass highlights all applicable peeps on pr creation as well as commit (if the somebody new needs to be notified)

**Comment by MrStonedOne** [comment]
Part of the disconnect is that people are conflicting requesting a review from somebody, with being notified in general.

**Comment by Shadowlight213** [comment]
@MrStonedOne we literally had a bot for a while that did that. I think it looked at who had modified the files touched in a pr last, and then mentioned them.

Add me to the meme for modular computer stuff.

**Comment by Cyberboss** [comment]
Read what I said above shadow. Also, I wasn't around for that, but from
what I gather it was involuntary tagging and people hated it.

**Comment by Shadowlight213** [comment]
@Cyberboss the bot has a config to disable looking at who touched the lines, and instead always only notify specific people for specific files. Basically exactly what the codeowners thing does.

**Comment by optimumtact** [comment]
@tgstation/commit-access someone merge this or I'm gonna close in 24 hours because I don't like it

**Comment by optimumtact** [comment]
1 hour remains @tgstation/commit-access

**Comment by optimumtact** [comment]
I'm giving this another 24 hours

**Comment by KorPhaeron** [comment]
Coward

**Comment by MoreRobustThanYou** [comment]
that's cute

**Comment by Shadowlight213** [comment]
Tbh, hg should be the only one in the codeowners file.
All code is his.

**Pull request closed without merging**